### PR TITLE
[2018-06] [sdks] Create archive targets for pre-building on CI

### PR DESCRIPTION
**Pull request by luhenry**
Backport of https://github.com/mono/mono/pull/10882

### CODE REVIEW

**Comment by luhenry** [comment]
@monojenkins build failed

File: mcs/build/rules.make [hunk 1]
```diff
@@ -14,10 +14,10 @@ _FILTER_OUT = $(foreach x,$(2),$(if $(findstring $(1),$(x)),,$(x)))
 # given $(thisdir), we compute the path to the top directory
 #
 # split_path = $(filter-out .,$(subst /,$(space),$(1)))
-# make_path = ./$(subst $(space),/,$(1))
 # dotdottify = $(patsubst %,..,$(1))
+# make_path = ./$(subst $(space),/,$(1))
 # topdir = $(call make_path,$(call dotdottify,$(call split_path,$(thisdir))))
-topdir := ./$(subst $(space),/,$(patsubst %,..,$(filter-out .,$(subst /,$(space),$(thisdir)))))
+topdir := $(abspath ./$(subst $(space),/,$(patsubst %,..,$(filter-out .,$(subst /,$(space),$(thisdir))))))
```
Comment (akoeplinger) [inline mcs/build/rules.make]: what exactly was malformed? I fought with these last week and abspath causes issues on Windows once you cross the cygwin/native divide

**Comment by luhenry** [comment]
@monojenkins build

**Comment by luhenry** [comment]
@akoeplinger it seems like we are still missing some commits to make that work.

**Pull request merged**